sechub-pds-examples/gosec/50-start-multiple-docker-compose.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: MIT
+
+SCRIPT_DIR=`dirname $0`
+REPLICAS="$1"
+
+ENVIRONMENT_FILE=".env-cluster"
+
```

Comment:
I started with `./50-start-multiple-docker-compose.sh 3`

after this I tried out with sechub to connect.

The problem here: SecHub adapter does not accept older TLS versions - the load balancer implementation seems to use old ones.

SecHub scanning mit single variant (01-... script) does work!

```
2021-09-02 14:43:53.711 ERROR 28739 --- [-7fbfd54100fb-2] .s.d.s.p.AbstractProductExecutionService : Product executor failed:PDS_CODESCAN __[SECHUB-UID:e374fa6e-bd5c-45ff-953e-7fbfd54100fb]__

com.daimler.sechub.sharedkernel.execution.SecHubExecutionException: PDS_CODESCAN execution failed.__[SECHUB-UID:e374fa6e-bd5c-45ff-953e-7fbfd54100fb]__
	at com.daimler.sechub.domain.scan.product.AbstractInstallSetupProductExecutor.execute(AbstractInstallSetupProductExecutor.java:85)
	at com.daimler.sechub.domain.scan.product.AbstractProductExecutionService.execute(AbstractProductExecutionService.java:93)
	at com.daimler.sechub.domain.scan.product.AbstractProductExecutionService.runOnExecutorWithOneConfiguration(AbstractProductExecutionService.java:187)
	at com.daimler.sechub.domain.scan.product.AbstractProductExecutionService.runOnAllAvailableExecutors(AbstractProductExecutionService.java:151)
	at com.daimler.sechub.domain.scan.product.AbstractProductExecutionService.executeProductsAndStoreResults(AbstractProductExecutionService.java:68)
	at com.daimler.sechub.domain.scan.ScanJobExecutor$CanceableScanJobRunnable.run(ScanJobExecutor.java:145)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.springframework.web.client.ResourceAccessException: I/O error on POST request for "https://localhost:8444/api/job/create": Received fatal alert: protocol_version; nested exception is javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:785)
	at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:711)
	at org.springframework.web.client.RestTemplate.postForEntity(RestTemplate.java:468)
	at com.daimler.sechub.adapter.support.RestOperationsSupport.postJSON(RestOperationsSupport.java:28)
	at com.daimler.sechub.adapter.pds.PDSAdapterV1.createNewPDSJOB(PDSAdapterV1.java:193)
	at com.daimler.sechub.adapter.pds.PDSAdapterV1.execute(PDSAdapterV1.java:46)
	at com.daimler.sechub.adapter.pds.DelegatingMockablePDSAdapterV1.execute(DelegatingMockablePDSAdapterV1.java:53)
	at com.daimler.sechub.adapter.pds.DelegatingMockablePDSAdapterV1.execute(DelegatingMockablePDSAdapterV1.java:1)
	at com.daimler.sechub.adapter.AbstractAdapter.start(AbstractAdapter.java:71)
	at com.daimler.sechub.domain.scan.product.pds.PDSCodeScanProductExecutor.lambda$0(PDSCodeScanProductExecutor.java:130)
	at com.daimler.sechub.sharedkernel.resilience.ResilientActionExecutor.executeResilient(ResilientActionExecutor.java:53)
	at com.daimler.sechub.sharedkernel.resilience.ResilientActionExecutor.executeResilient(ResilientActionExecutor.java:39)
	at com.daimler.sechub.domain.scan.product.pds.PDSCodeScanProductExecutor.executeWithAdapter(PDSCodeScanProductExecutor.java:94)
	at com.daimler.sechub.domain.scan.product.pds.PDSCodeScanProductExecutor.executeWithAdapter(PDSCodeScanProductExecutor.java:1)
	at com.daimler.sechub.domain.scan.product.AbstractInstallSetupProductExecutor.executeAdapterWhenTargetTypeSupported(AbstractInstallSetupProductExecutor.java:191)
	at com.daimler.sechub.domain.scan.product.AbstractInstallSetupProductExecutor.execute(AbstractInstallSetupProductExecutor.java:165)
	at com.daimler.sechub.domain.scan.product.AbstractInstallSetupProductExecutor.execute(AbstractInstallSetupProductExecutor.java:77)
	... 6 common frames omitted
Caused by: javax.net.ssl.SSLHandshakeException: Received fatal alert: protocol_version
	at sun.security.ssl.Alert.createSSLException(Alert.java:131)
	at sun.security.ssl.Alert.createSSLException(Alert.java:117)
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:311)
	at sun.security.ssl.Alert$AlertConsumer.consume(Alert.java:293)
	at sun.security.ssl.TransportContext.dispatch(TransportContext.java:185)
	at sun.security.ssl.SSLTransport.decode(SSLTransport.java:149)
	at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1143)
	at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1054)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:394)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436)
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:374)
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
	at org.springframework.http.client.HttpComponentsClientHttpRequest.executeInternal(HttpComponentsClientHttpRequest.java:87)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:66)
	at org.springframework.http.client.BufferingClientHttpRequestWrapper.executeInternal(BufferingClientHttpRequestWrapper.java:63)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:66)
	at org.springframework.http.client.InterceptingClientHttpRequest$InterceptingRequestExecution.execute(InterceptingClientHttpRequest.java:109)
	at com.daimler.sechub.adapter.pds.PDSClientHttpRequestInterceptor.intercept(PDSClientHttpRequestInterceptor.java:36)
	at org.springframework.http.client.InterceptingClientHttpRequest$InterceptingRequestExecution.execute(InterceptingClientHttpRequest.java:93)
	at com.daimler.sechub.adapter.TraceLogClientHTTPRequestInterceptor.intercept(TraceLogClientHTTPRequestInterceptor.java:32)
	at org.springframework.http.client.InterceptingClientHttpRequest$InterceptingRequestExecution.execute(InterceptingClientHttpRequest.java:93)
	at org.springframework.http.client.InterceptingClientHttpRequest.executeInternal(InterceptingClientHttpRequest.java:77)
	at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48)
	at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:66)
	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:776)
	... 22 common frames omitted
```

So load balancer mechanism /TLS forwarding or 
maybe nginx configuration must be changed.

Reply:
Nginx is configured to use TLSv1.3 only (see: [docker/loadbalancer/nginx.conf  -> line 22](https://github.com/Daimler/sechub/pull/696/files#diff-a2e5a11f90e2d3be3051ff5a20065c15caaaba502db17b91044db9986a6bbc00R22))

Reply:
By setting `-Djavax.net.debug=all` one can get more details of the TLS connection. For more information have a look at: https://docs.oracle.com/javase/7/docs/technotes/guides/security/jsse/ReadDebug.html.

Reply:
For more information about the TLS connection use the OpenSSL client: `openssl s_client -connect localhost:8444`

Reply:
After enabling  `-Djavax.net.debug=all` it becomes clear, that the PDS server uses TLS 1.2 to negotiate the connection:

~~~
javax.net.ssl|FINE|4A|SecHub-exec-f40414fa-b43f-4caa-abf3-8822df8ebb0d-1|2021-09-08 20:12:18.122 CEST|ClientHello.java:575|Produced ClientHello handshake message (
"ClientHello": {
  "client version"      : "TLSv1.2",
  "random"              : "4F 48 AD C0 6C 85 7E 30 ED 0E 9B BA E9 C6 33 AE 24 64 38 DB A8 05 6B C3 03 94 29 A7 C9 6D 01 58",
  "session id"          : "",
  "cipher suites"       : "[TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384(0xC02C), TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256(0xC02B), TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384(0xC030), TLS_RSA_WITH_AES_256_GCM_SHA384(0x009D), TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384(0xC02E), TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384(0xC032), TLS_DHE_RSA_WITH_AES_256_GCM_SHA384(0x009F), TLS_DHE_DSS_WITH_AES_256_GCM_SHA384(0x00A3), TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256(0xC02F), TLS_RSA_WITH_AES_128_GCM_SHA256(0x009C), TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256(0xC02D), TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256(0xC031), TLS_DHE_RSA_WITH_AES_128_GCM_SHA256(0x009E), TLS_DHE_DSS_WITH_AES_128_GCM_SHA256(0x00A2), TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384(0xC024), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384(0xC028), TLS_RSA_WITH_AES_256_CBC_SHA256(0x003D), TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384(0xC026), TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384(0xC02A), TLS_DHE_RSA_WITH_AES_256_CBC_SHA256(0x006B), TLS_DHE_DSS_WITH_AES_256_CBC_SHA256(0x006A), TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA(0xC00A), TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA(0xC014), TLS_RSA_WITH_AES_256_CBC_SHA(0x0035), TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA(0xC005), TLS_ECDH_RSA_WITH_AES_256_CBC_SHA(0xC00F), TLS_DHE_RSA_WITH_AES_256_CBC_SHA(0x0039), TLS_DHE_DSS_WITH_AES_256_CBC_SHA(0x0038), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256(0xC023), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256(0xC027), TLS_RSA_WITH_AES_128_CBC_SHA256(0x003C), TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256(0xC025), TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256(0xC029), TLS_DHE_RSA_WITH_AES_128_CBC_SHA256(0x0067), TLS_DHE_DSS_WITH_AES_128_CBC_SHA256(0x0040), TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA(0xC009), TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA(0xC013), TLS_RSA_WITH_AES_128_CBC_SHA(0x002F), TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA(0xC004), TLS_ECDH_RSA_WITH_AES_128_CBC_SHA(0xC00E), TLS_DHE_RSA_WITH_AES_128_CBC_SHA(0x0033), TLS_DHE_DSS_WITH_AES_128_CBC_SHA(0x0032), TLS_EMPTY_RENEGOTIATION_INFO_SCSV(0x00FF)]",
  "compression methods" : "00",
  "extensions"          : [
    "supported_groups (10)": {
      "versions": [secp256r1, secp384r1, secp521r1, ffdhe2048, ffdhe3072, ffdhe4096, ffdhe6144, ffdhe8192]
    },
    "ec_point_formats (11)": {
      "formats": [uncompressed]
    },
    "signature_algorithms (13)": {
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, dsa_sha256, ecdsa_sha224, rsa_sha224, dsa_sha224, ecdsa_sha1, rsa_pkcs1_sha1, dsa_sha1]
    },
    "signature_algorithms_cert (50)": {
      "signature schemes": [ecdsa_secp256r1_sha256, ecdsa_secp384r1_sha384, ecdsa_secp521r1_sha512, rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512, rsa_pss_pss_sha256, rsa_pss_pss_sha384, rsa_pss_pss_sha512, rsa_pkcs1_sha256, rsa_pkcs1_sha384, rsa_pkcs1_sha512, dsa_sha256, ecdsa_sha224, rsa_sha224, dsa_sha224, ecdsa_sha1, rsa_pkcs1_sha1, dsa_sha1]
    },
    "extended_master_secret (23)": {
      <empty>
    },
    "supported_versions (43)": {
      "versions": [TLSv1.2]
    }
  ]
}
)
~~~

However, Nginx uses TLSv1.3 only.

Reply:
The solution is to add TLSv1.2 support for Ngnix.

Reply:
**Just for documentation:**
We have inspected the reason why we need TLSv1.2 here:
When using Java 8 (even the latest one), TLSv1.3 is NOT enabled per default for client operations.
Please refer https://www.oracle.com/java/technologies/javase/8u261-relnotes.html for details.

So when starting SecHub server with Java8 the adapter will only be capable to connect via TLSv1.2 which is accepted by server side with correct ciphers and no SHA1 etc. (see https://github.com/Daimler/sechub/issues/625)